### ISSUE TITLE
Provision Redis in BSP environments

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#1537]
## User Story

As a Data.gov operator, I want Redis provisioned in the BSP environments so that I can meet the dependencies of CKAN 2.8 without taking on additional infrastructure burden.

## Acceptance Criteria
- [x] GIVEN I run the Inventory playbook
  WHEN I inspect the ckan.harvester.mq setting /etc/ckan/production.ini on Inventory hosts
  THEN I see CKAN is configured to connect to the new Redis host(s).

## Details

~We want to hand-off a terraform template (https://github.com/GSA/datagov-deploy/issues/1536) to BSP so they know exactly how to configure Redis ElastiCache. https://gist.github.com/adborden/1f6ccddcf1b4f5f87d6fb1a737e142ef~

We'll go forward with a dedicated redis host instead of ElastiCache due to constraints in BSP.

**Comment by adborden** [comment]
I sent an email to BSP yesterday to see if ElastiCache is an option.

**Comment by mogul** [comment]
@adborden can you ping that thread?

**Comment by adborden** [comment]
Pinged.

**Comment by mogul** [comment]
Still waiting on an answer here. Hyon is including this languishing query in a round-up she's talking to FCS mgmt about.

**Comment by adborden** [comment]
tl;dr we'll go forward with a self-hosted redis instead of ElastiCache. I've updated the AC and will create a new BSP ticket.

BSP got back to us. They _could_ provide ElastiCache but they are concerned that without console or CLI access we'll be too dependent on them, which makes sense to us.

**Comment by adborden** [comment]
Opened a ticket for BSP and will reach out in chat.

```
Please create a new stack based on Ubuntu 18.04 for a stand-alone Redis instance for Data.gov
AMI: based on Ubuntu 18.04
instance type: t3.xlarge or m5.xlarge (looking for some advice here)
name template: redis1p
EBS:
- additional 16GB ebs volume for data
- additional 16GB volume for /var/log

Once the stack is created, please create an instance in DEV and one in PROD.

We currently use catalog-harvester1p for Redis, so we can use that as a basis for any other configuration.
```

**Comment by adborden** [comment]
BSP provided some helpful information on the instance sizing:

> I see that you are looking to use t3.xlarge or m5.xlarge. I'm not sure what your use case is or if this is a cost consideration but first difference I'll point out is the difference in network performance. If that's not an issue, I'd consider the following from AWS:
>
> When to use unlimited mode versus fixed CPU
> When determining whether you should use a burstable performance instance in unlimited mode, such as a T3, or a fixed performance instance, such as an M5, you need to determine the breakeven CPU usage. The breakeven CPU usage for a burstable performance instance is the point at which a burstable performance instance costs the same as a fixed performance instance. The breakeven CPU usage helps you determine the following:
>
>If the average CPU usage over a 24-hour period is at or below the breakeven CPU usage, use a burstable performance instance in unlimited mode so that you can benefit from the lower price of a burstable performance instance while getting the same performance as a fixed performance instance.
>
> If the average CPU usage over a 24-hour period is above the breakeven CPU usage, the burstable performance instance will cost more than the equivalently-sized fixed performance instance. If a T3 instance continuously bursts at 100% CPU, you end up paying approximately 1.5 times the price of an equivalently-sized M5 instance.

**Comment by adborden** [comment]
T3 supports up to 5Gbps, which should be sufficient. To estimate the CPU load, I'm looking at catalog-harvester1p, which currently hosts both Redis and the Harvester workers themselves. The new instance will only host Redis, so this serves as an overestimate of CPU load. 

![Screenshot from 2020-06-19 10-44-38](https://user-images.githubusercontent.com/509703/85166116-b04e3d00-b21b-11ea-8bf3-2588f7ca9b15.png)

Over the course of 7 days, we never see the average go above 5.5%, so I think a T3 will make the most sense.

... and fwiw, all our instances are likely sized incorrectly.

**Comment by adborden** [comment]
AWS documents the [breakeven CPU usage for t3.xlarge](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/burstable-performance-instances-unlimited-mode-concepts.html) at 52.5%.

**Comment by mogul** [comment]
Per meeting yesterday, we're still waiting on the FCS side here. It's not expected to be a lot of work, just hasn't yet reached the top of their pile. We're hoping we'll be unblocked during our upcoming sprint.

**Comment by adborden** [comment]
Had to send the instances back to BSP, the drive configuration is wrong, should be:
- 8GB root /
- 16GB data /data
- 16GB logs /var/log

Also, the wrong SSH key was used for redis1p.

**Comment by mogul** [comment]
BSP resolved the incorrect configuration. Should be resolvable in short order.
